src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2234,5 +2258,28 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 2;
             return true;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsAllZero(this Vector128<byte> vector)
```

Comment:
nit: My vote is we don't add these.

The `TestZ` optimization is already being done on x86/x64: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L1251-L1290

`==` is likewise optimized to fold additional instructions away and better optimize the code "in general" depending on whether the comparison is used as a value or a branch.

The same applies for ARM64 (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerarmarch.cpp#L784) modulo that it doesn't additionally optimize down to "just" `MaxAcross` in the case you are comparing against `zero`. This is a minor and temporary two instruction cost that can be tracked for .NET 7 as part of the general ARM64 push.

Reply:
Makes sense, removed!